dspace-api/src/main/java/org/dspace/content/dao/impl/CollectionDAOImpl.java

```diff
@@ -75,29 +81,35 @@ protected CollectionDAOImpl() {
 
     @Override
     public Collection findByTemplateItem(Context context, Item item) throws SQLException {
-        Criteria criteria = createCriteria(context, Collection.class);
-        criteria.add(Restrictions.eq("template_item", item));
-        return uniqueResult(criteria);
+        CriteriaBuilder criteriaBuilder = getCriteriaBuilder(context);
+        CriteriaQuery criteriaQuery = getCriteriaQuery(criteriaBuilder, Collection.class);
+        Root<Collection> collectionRoot = criteriaQuery.from(Collection.class);
+        //TODO Used to be template_item, may be wrong
```

Comment:
RE: this TODO. Do we need it?  Looks like the old query did "template_item" = item.   Now, the new one looks to be doing "template_item_id" = item.  I think it likely is correct.  Should we remove this TODO?